### ISSUE TITLE
DistributedOptimzer is not compatible with keras.Optimizer

### ISSUE THREAD

**Issue by yselivonchyk** [horovod/horovod#1437]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version: 1.14
3. Horovod version: 0.16.4
7. Python version: 3.6.8

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Bug report:**
Please describe errorneous behavior you're observing and steps to reproduce it.
Horovod DistributedOptimzer wrapper is not compatible with keras:

```
import tensorflow as tf
import horovod.tensorflow.keras as hvd

hvd.init()
opt = tf.keras.optimizers.Adam()
hopt = hvd.DistributedOptimizer(opt)
opt.get_config()
cfg = hopt.get_config()
opt_copy = opt.from_config(cfg)
opt_copy = opt.__class__.from_config(cfg)
hopt_copy = hopt.from_config(cfg) # TypeError: __init__() got an unexpected keyword argument 'learning_rate'
hopt_copy = hopt.__class__.from_config(cfg) # TypeError: __init__() got an unexpected keyword argument 'learning_rate'
```

**Comment by tgaddair** [comment]
Hey @yselivonchyk, for using `tf.keras` can you try importing `horovod.tensorflow.keras` instead of `horovod.tensorflow`?

**Comment by yselivonchyk** [comment]
@tgaddair Thank you for pointing that out. I updated the issue to reflect how hvd.tensorflow.keras is not compatible with tf.keras.optimizers.Optimizer.

**Comment by yselivonchyk** [comment]
To put it in context, this functionality is used by `tensorflow.python.keras.models.clone_and_build_model` which is called by` tf.keras.experimental.export_saved_model`

**Comment by tgaddair** [comment]
Hey @yselivonchyk, for the new example, can you try this instead:

```
import tensorflow as tf
import horovod.tensorflow.keras as hvd

hvd.init()
opt = tf.keras.optimizers.Adam()
hopt = hvd.DistributedOptimizer(opt)
opt.get_config()
cfg = hopt.get_config()
opt_copy = opt.from_config(cfg)
opt_copy = opt.__class__.from_config(cfg)
hopt_copy = hvd.DistributedOptimizer(opt_copy)
```

Basically, we never directly instantiate the `DistributedOptimizer`, we always create it by wrapping another optimizer.

**Comment by yselivonchyk** [comment]
It would not solve the problem of `.from_config(cfg)` not working and  `tensorflow.python.keras.models.clone_and_build_model` along with it.

**Comment by tgaddair** [comment]
I see, thanks for clarifying the usage @yselivonchyk.  That makes sense.  I put together a quick PR with your suggested fix (#1444).  Feel free to try it out and let me know if it solves the issue.
